pkg/apis/core/v1beta1/helper/errors.go

```diff
@@ -52,16 +52,36 @@ var (
 	dependenciesRegexp           = regexp.MustCompile(`(?i)(PendingVerification|Access Not Configured|accessNotConfigured|DependencyViolation|OptInRequired|DeleteConflict|Conflict|inactive billing state|ReadOnlyDisabledSubscription|is already being used|not available in the current hardware cluster)`)
 )
 
-// DetermineError determines the Garden error code for the given error message.
-func DetermineError(message string) error {
-	code := determineErrorCode(message)
-	if code == "" {
+// DetermineError determines the Garden error code for the given error and creates a new error with the given message.
+func DetermineError(err error, message string) error {
+	if err == nil {
 		return errors.New(message)
 	}
-	return &errorWithCode{code, message}
+
+	errMsg := message
+	if errMsg == "" {
+		errMsg = err.Error()
+	}
```

Comment:
```suggestion
	if message == "" {
		message = err.Error()
	}
```

Reply:
Hm, why do you want to work on an argument variable? I've never seen this being considered as best practice 👀